backend/app/routes/admin/resetPassword.js

```diff
@@ -0,0 +1,52 @@
+const argon2 = require('argon2');
+const Admin = require('../../models/Admin');
+const { ErrorHandler } = require('../../../helpers/error');
+const constants = require('../../../constants');
+const { verifyToken } = require('../../../helpers/middlewares/auth');
+
+module.exports = async (req, res, next) => {
+  // Extracting the new password from the request
+  const { newPassword } = req.body;
+  try {
```

Comment:
I think so we can exclude this try catch block since you are not switching over the different kinds of error. Instead try handling the error cases without this. You can use await-to-js for async operations

Reply:
As this is an async function there is no harm using `try and catch` here and it also looks syntactically clean the main reason for using try and catch is because of the `verifyToken` function which throws an error if the token is not valid so the catch block will catch that and send a response accordingly.

Reply:
Yes, for a neater experience with additional benefits we are using await-to-js which comes with similar error handling without explicitly using try catch